packages/widget-v2/src/components/MainButton.tsx

```diff
@@ -0,0 +1,165 @@
+import { styled } from 'styled-components';
+import { Row } from './Layout';
+import { SmallText, Text } from './Typography';
+import { PlusIcon } from '../icons/PlusIcon';
+import { useTheme } from 'styled-components';
+import { CheckmarkIcon } from '../icons/CheckmarkIcon';
+import { LeftArrowIcon, RightArrowIcon } from '../icons/ArrowIcon';
+import { SwapIcon } from '../icons/SwapIcon';
+import { WarningIcon } from '../icons/WarningIcon';
+
+export enum ICONS {
+  plus,
+  checkmark,
+  rightArrow,
+  leftArrow,
+  swap,
+  warning,
+}
+
+const iconMap = {
+  [ICONS.plus]: PlusIcon,
+  [ICONS.checkmark]: CheckmarkIcon,
+  [ICONS.rightArrow]: RightArrowIcon,
+  [ICONS.leftArrow]: LeftArrowIcon,
+  [ICONS.swap]: SwapIcon,
+  [ICONS.warning]: WarningIcon,
+  none: () => null,
+};
+
+export type MainButtonProps = {
+  label: string;
+  icon?: ICONS | 'none';
+  leftIcon?: ICONS | 'none';
+  backgroundColor?: string;
+  disabled?: boolean;
+  loading?: boolean;
+  loadingTimeString?: string;
+};
+
+export const MainButton = ({
+  label,
+  icon = 'none',
+  leftIcon = 'none',
+  backgroundColor,
+  disabled,
+  loading,
+  loadingTimeString,
+}: MainButtonProps) => {
+  const theme = useTheme();
+  backgroundColor ??= disabled ? theme.secondary.background : theme.brandColor;
+
+  const Icon = iconMap[icon];
+  const LeftIcon = iconMap[leftIcon];
+
+  if (loading) {
+    return (
+      <StyledMainButton
+        align="center"
+        justify="center"
+        backgroundColor={theme.backgroundColor}
+        loading={loading}
+      >
+        <StyledOverlay
+          className="overlay"
+          justify="space-between"
+          padding={20}
+          backgroundColor={theme.backgroundColor}
+        >
+          <Text fontSize={24} style={{ opacity: 0.5 }}>
+            {label}
+          </Text>
+          {loadingTimeString && (
+            <StyledTimeRemaining align="center" justify="center">
+              <SmallText>{loadingTimeString}</SmallText>
+            </StyledTimeRemaining>
+          )}
+        </StyledOverlay>
+      </StyledMainButton>
+    );
+  }
```

Comment:
This loading animation is not perfect, not sure if we want to switch to using something better but I'm gonna leave it like this for now